packages/hash/frontend/src/pages/[account-slug]/types/entity-type/property-type-form.tsx

```diff
@@ -1,60 +1,365 @@
 
+type PropertyTypeFormValues = {
+  name: string;
+  description: string;
+  expectedValues: typeof types["dataType"][PrimitiveDataTypeKey][];
+};
+
+const generateInitialPropertyTypeId = (baseUri: string) =>
+  addVersionToBaseUri(baseUri, 1);
+
+const faText: IconDefinition["icon"] = [
+  448,
+  512,
+  [],
+  "f893",
+  "M448 56v80C448 149.3 437.3 160 424 160S400 149.3 400 136V80h-152v352h48c13.25 0 24 10.75 24 24S309.3 480 296 480h-144C138.8 480 128 469.3 128 456s10.75-24 24-24h48v-352H48v56C48 149.3 37.25 160 24 160S0 149.3 0 136v-80C0 42.75 10.75 32 24 32h400C437.3 32 448 42.75 448 56z",
+];
+
+const fa100: IconDefinition["icon"] = [
+  512,
+  512,
+  [],
+  "e41c",
+  "M171.2 99.64C175.3 61.17 207.8 32 246.5 32C291.5 32 326.5 71.02 321.7 115.8L308.8 236.4C304.7 274.8 272.2 304 233.5 304C188.5 304 153.5 264.1 158.3 220.2L171.2 99.64zM246.5 80C232.3 80 220.4 90.68 218.9 104.8L205.1 225.3C204.2 241.7 217.1 256 233.5 256C247.7 256 259.6 245.3 261.1 231.2L274 110.7C275.8 94.28 262.9 80 246.5 80V80zM118.4 36.59C125.3 41.65 128.9 50.04 127.8 58.59L95.82 314.6C94.17 327.8 82.18 337.3 69.02 335.9C55.87 334.4 46.54 322.6 48.19 309.4L74.99 94.99L41.45 109.9C29.26 115.3 15.15 109.9 9.938 97.81C4.721 85.71 10.37 71.52 22.56 66.1L94.56 34.1C102.5 30.59 111.6 31.54 118.4 36.59V36.59zM507.2 114.1L499.9 202.5C496.6 241.8 463.8 272 424.3 272C379.1 272 345.1 234.1 348.8 189.9L356.1 101.5C359.4 62.23 392.3 32 431.7 32C476 32 510.9 69.91 507.2 114.1V114.1zM396.6 193.9C395.2 210.1 408 224 424.3 224C438.8 224 450.8 212.9 452 198.5L459.4 110.1C460.8 93.91 447.1 79.1 431.7 79.1C417.2 79.1 405.2 91.09 403.1 105.5L396.6 193.9zM511.7 323.9C513.9 336.1 505.1 349.4 492.1 351.7L28.08 431.7C15.02 433.9 2.605 425.1 .353 412.1C-1.899 399 6.864 386.6 19.93 384.3L483.9 304.3C496.1 302.1 509.4 310.9 511.7 323.9H511.7zM219.9 479.7C206.9 481.9 194.5 473 192.3 459.9C190.2 446.9 198.1 434.5 212.1 432.3L452.1 392.3C465.1 390.1 477.5 398.1 479.7 412.1C481.9 425.1 473 437.5 459.9 439.7L219.9 479.7z",
+];
+
+const faSquareCheck: IconDefinition["icon"] = [
+  448,
+  512,
+  [],
+  "f14a",
+  "M211.8 339.8C200.9 350.7 183.1 350.7 172.2 339.8L108.2 275.8C97.27 264.9 97.27 247.1 108.2 236.2C119.1 225.3 136.9 225.3 147.8 236.2L192 280.4L300.2 172.2C311.1 161.3 328.9 161.3 339.8 172.2C350.7 183.1 350.7 200.9 339.8 211.8L211.8 339.8zM0 96C0 60.65 28.65 32 64 32H384C419.3 32 448 60.65 448 96V416C448 451.3 419.3 480 384 480H64C28.65 480 0 451.3 0 416V96zM48 96V416C48 424.8 55.16 432 64 432H384C392.8 432 400 424.8 400 416V96C400 87.16 392.8 80 384 80H64C55.16 80 48 87.16 48 96z",
+];
+
+const propertyTypeDataTypes = [
+  {
+    title: types.dataType.text.title,
+    icon: faText,
+    dataTypeId: types.dataType.text.dataTypeId,
+  },
+  {
+    title: types.dataType.number.title,
+    icon: fa100,
+    dataTypeId: types.dataType.number.dataTypeId,
+  },
+  {
+    title: types.dataType.boolean.title,
+    icon: faSquareCheck,
+    dataTypeId: types.dataType.boolean.dataTypeId,
+  },
+];
```

Comment:
I guess we'll be adding more of these icons (since there will be more types based on design? ), we could move the icons to a separate file to keep this one shorted (it's already really long - 365lines).

Reply:
[`bf93dca` (#1345)](https://github.com/hashintel/hash/pull/1345/commits/bf93dca072b272643ed76c46882b3fd21e701156)